pkg/mapi2capi/machine.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mapi2capi
+
+import (
+	mapiv1 "github.com/openshift/api/machine/v1beta1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	capiNamespace            = "openshift-cluster-api"
+	workerUserDataSecretName = "worker-user-data"
+	awsTemplateAPIVersion    = "infrastructure.cluster.x-k8s.io/v1beta2"
+	awsTemplateKind          = "AWSMachineTemplate"
+)
+
+// fromMachineToMachine translates a MAPI Machine to its Core CAPI Machine correspondent.
+func fromMachineToMachine(mapiMachine *mapiv1.Machine) capiv1.Machine {
+	capiMachine := capiv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        mapiMachine.Name,
+			Namespace:   capiNamespace,
+			Labels:      mapiMachine.Labels,
+			Annotations: mapiMachine.Annotations,
+		},
+		Spec: capiv1.MachineSpec{
+			InfrastructureRef: corev1.ObjectReference{
+				APIVersion: awsTemplateAPIVersion,
+				Kind:       awsTemplateKind,
+				Name:       mapiMachine.Name,
+			},
+			ProviderID: mapiMachine.Spec.ProviderID,
+
+			// Version: not necessary (optionally used by bootstrap providers).
+			// FailureDomain: populated by higher level functions.
+			// ClusterName: populated by higher level functions.
+
+			// TODO: figure out if/how to convert these.
+			// NodeDrainTimeout: ,
+			// NodeVolumeDetachTimeout: ,
+			// NodeDeletionTimeout: ,
```

Comment:
How do we deal with fields that exist only within MAPI here, eg, I would expect an error presently for a MAPI machine that has taints, but I don't see where that is

Reply:
Yes errors, see the Taints handling I added now, we can use that as an example.